manager/integration/tests/test_recurring_job.py

```diff
@@ -1239,100 +1224,199 @@ def test_recurring_job_delete(client, batch_v1_beta_api, volume_name):  # NOQA
-        add `backup`   recurring job to volume label.
-    And `snapshot1` cron job exist.
-        `backup1`   cron job exist.
-
-    When delete `snapshot` recurring job.
-         delete `backup`   recurring job.
-    Then `snapshot` cron job should not exist.
-         `backup`   cron job should not exist.
-    And `snapshot` job  should exist in volume recurring job label.
-        `backup`   job  should exist in volume recurring job label.
-        `default` group should exist in volume recurring job label.
-
-    # Add back the recurring jobs.
-    When create `snapshot` recurring job.
-         create `backup`   recurring job.
-    Then `snapshot` cron job should exist.
-         `backup`   cron job should exist.
-    """
+    # Delete recurring job-group in-use by other recurring-job
+    # should not remove the volume label
     recurring_jobs = {
-        SNAPSHOT: {
-            TASK: SNAPSHOT,
-            GROUPS: [],
-            CRON: SCHEDULE_1MIN,
-            RETAIN: 1,
-            CONCURRENCY: 2,
-            LABELS: {},
-        },
-        BACKUP: {
-            TASK: BACKUP,
-            GROUPS: [],
-            CRON: SCHEDULE_1MIN,
-            RETAIN: 1,
-            CONCURRENCY: 2,
-            LABELS: {},
-        },
+        snap1: snap_job,
+        snap2: snap_job,
+        back1: back_job,
     }
     create_recurring_jobs(client, recurring_jobs)
     check_recurring_jobs(client, recurring_jobs)
 
-    client.create_volume(name=volume_name, size=SIZE)
-    volume = wait_for_volume_detached(client, volume_name)
-    volume.attach(hostId=get_self_host_id())
-    volume = wait_for_volume_healthy(client, volume_name)
-    volume.recurringJobAdd(name=SNAPSHOT, isGroup=False)
-    volume.recurringJobAdd(name=BACKUP, isGroup=False)
-    wait_for_cron_job_create(batch_v1_beta_api, JOB_LABEL+"="+SNAPSHOT)
-    wait_for_cron_job_create(batch_v1_beta_api, JOB_LABEL+"="+BACKUP)
+    volume.recurringJobAdd(name=group1, isGroup=True)
+    wait_for_volume_recurring_job_update(volume,
+                                         jobs=[], groups=[group1, DEFAULT])
+    volume.recurringJobDelete(name=DEFAULT, isGroup=True)
+    wait_for_volume_recurring_job_update(volume,
+                                         jobs=[], groups=[group1])
 
-    snap1_recurring_job = client.by_id_recurring_job(SNAPSHOT)
-    back1_recurring_job = client.by_id_recurring_job(BACKUP)
+    snap1_recurring_job = client.by_id_recurring_job(snap1)
     client.delete(snap1_recurring_job)
+    time.sleep(5)
+    wait_for_volume_recurring_job_update(volume,
+                                         jobs=[], groups=[group1])
+
+    snap2_recurring_job = client.by_id_recurring_job(snap2)
+    client.delete(snap2_recurring_job)
+    time.sleep(5)
+    wait_for_volume_recurring_job_update(volume,
+                                         jobs=[], groups=[group1])
+
+    # Delete last recurring job in group remove the volume label
```

Comment:
Do you mean `Deleting last recurring job of the group would lead to the group label being cleaned up for all volumes`? We need to add this warning message/tip in Longhorn UI.
Does this logic apply to the default group?

Reply:
> Do you mean `Deleting last recurring job of the group would lead to the group label being cleaned up for all volumes`?

Reworded.

> We need to add this warning message/tip in Longhorn UI. 
Sure I will create an issue.

>Does this logic apply to the default group?
Should not affect the default group auto-assignment. I've added steps to the test case.

Reply:
https://github.com/longhorn/longhorn/issues/3651